crates/anvil-polkadot/src/substrate_node/host.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) 2023 Polytope Labs (Caymans) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Host function overrides for signature verification.
```

Comment:
let's add a comment in this file explaining why both overrides for key recovery and hashing are needed and how they work together